.github/workflows/tests.yml

```diff
@@ -265,33 +265,52 @@ jobs:
       - run: make slim-builds
       - run: make test-integration-splunk
 
-  test-integration-kubernetes:
-    name: Integration - Linux, Kubernetes, flaky
-    # This is an experimental test. Allow it to fail without failing the whole
-    # workflow, but keep it executing on every build to gather stats.
-    continue-on-error: true
+  test-e2e-kubernetes:
+    name: E2E - K8s ${{ matrix.kubernetes_version }} / ${{ matrix.container_runtime }}
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        kubernetes:
-          - v1.18.2
-          - v1.17.5
-          - v1.16.9
-          - v1.15.11
-          - v1.14.10
+        minikube_version:
+          - 'v1.11.0' # https://github.com/kubernetes/minikube/issues/8799
+        kubernetes_version:
```

Comment:
So this is... 5 * 3 test runners **just** for k8s?

Reply:
Well, it was worth it during development - helped to find a couple of interesting bugs along the way in unexpected places, that were only present in non-trivial combinations.
I don't mind reducing it, but it's not clear what subset of those is reasonable.

On a side note, there are a lot of ways we could optimize the E2E tests in general, to make them run quicker so that they don't create as much load on the system. Technically, the "payload" of these tests is very lightweight, and the most time, yet again, is spent during env prep and build. If we could share build artifacts - it would be great.

I feel like if we optimize the process in general we won't have to worry about having 15 tests...

We could move them into a separate workflow though, such that they'll live under their own section.

Reply:
Yeah this would consume actually 15% of our total possible github actions runners so we should consider the value here very, very carefully.

Reply:
Right. Well, we can exclude some of the matrix incarnations if we start facing issues... Just need to figure out which ones are the safest. Alternatively, this can be transformed into a single sequential invocation - but I'd hold on with that until we hit an actual problem.